compiler/rustc_expand/src/mbe/macro_check.rs

```diff
@@ -335,10 +335,17 @@ fn check_occurrences(
         }
         TokenTree::MetaVar(span, name) => {
             let name = MacroRulesNormalizedIdent::new(name);
-            check_ops_is_prefix(sess, node_id, macros, binders, ops, span, name);
+            check_ops_is_prefix(sess, node_id, macros, binders, ops, span, Some(name));
+        }
+        TokenTree::MetaVarExpr(dl, ref mve) => {
+            let name_opt = match mve {
```

Comment:
```suggestion
            let name_opt = mve.ident().map(MacroRulesNormalizedIdent::new);
```
Also, `MetaVarExpr::ident` needs to return `Option<Ident>` instead of `Option<&Ident>`, I didn't notice it in the previous PR.